src/repo/fs.rs

```diff
@@ -89,3 +90,214 @@ pub fn create_file<P: AsRef<Path>>(path: P) -> io::Result<File> {
         Ok(file) => Ok(file),
     }
 }
+
+#[cfg(unix)]
+const OS_ERROR_DIR_NOT_EMPTY: i32 = 39 /* ENOTEMPTY */;
+#[cfg(windows)]
+const OS_ERROR_DIR_NOT_EMPTY: i32 = 145 /* ERROR_DIR_NOT_EMPTY */;
+
+/// Removes empty directories by starting at [`leaf_dir`] and bubbling up
+/// until `boundary_dir` is reached.
+///
+/// # Arguments
+/// * `leaf_dir` - the leaf directory from which to start the removal
+/// * `boundary_dir` - stops when this directory is reached
+pub fn remove_empty_dirs<P1: AsRef<Path>, P2: AsRef<Path>>(
+    leaf_dir: P1,
+    boundary_dir: P2,
+) -> io::Result<()> {
+    assert!(
+        Iterator::eq(
+            leaf_dir
+                .as_ref()
+                .components()
+                .take(boundary_dir.as_ref().components().count()),
+            boundary_dir.as_ref().components()
+        ),
+        "leaf_dir must be a sub-directory of boundary_dir"
```

Comment:
is this logically the same as 'starts with', is there a function that can be used?